schema.json

```diff
@@ -0,0 +1,488 @@
+{
+    "$schema": "https://json-schema.org/draft-07/schema",
+    "type": "object",
+    "properties": {
+        "copyright": {
+            "type": "string",
+            "description": "The license string to include at the top of the generated header"
+        },
+        "enum_prefix": {
+            "type": "string",
+            "description": "The dedicated enum prefix for the implementation specific header to avoid collisions"
+        },
+        "constants": {
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    "name": {
+                        "type": "string",
+                        "description": "Name of the constant variable/define"
```

Comment:
(Can be future work)
Maybe we could use regex patterns to validate these a little more?
https://json-schema.org/understanding-json-schema/reference/string#regexp

These would get peppered all over the place so we'd probably want to link to the definition by reference:
https://stackoverflow.com/a/62585173/1268131